experimenter/experimenter/nimbus_ui_new/forms.py

```diff
@@ -35,6 +36,58 @@ def save(self, *args, **kwargs):
         return experiment
 
 
+class NimbusSerializerErrorMixin:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validation_errors = {}
+
+    def clean(self):
+        cleaned_data = super().clean()
+        if self.show_validation_errors:
+            for key, value in cleaned_data.items():
+                if isinstance(value, models.QuerySet):
+                    cleaned_data[key] = [item.pk for item in value]
+
+            for field_name in [
+                "excluded_experiments_branches",
+                "required_experiments_branches",
+            ]:
+                if field_name in cleaned_data:
+                    slugs = [
+                        slug_and_branch.split(":")[0]
+                        for slug_and_branch in cleaned_data[field_name]
+                    ]
+                    pks = list(
+                        NimbusExperiment.objects.filter(slug__in=slugs).values_list(
+                            "pk", flat=True
+                        )
+                    )
+                    cleaned_data[field_name.replace("_branches", "")] = pks
+
+            errors = self.instance.get_invalid_fields_errors(data=cleaned_data)
+
+            for field, error in errors:
+                if field in self.fields:
+                    if field not in self.validation_errors:
+                        self.validation_errors[field] = []
+                    self.validation_errors[field].append(error)
+
+            return cleaned_data
+
+        return cleaned_data
+
+
+class ShowValidationErrorsMixin:
+    def __init__(self, *args, **kwargs):
+        if self.request:
+            self.show_validation_errors = (
+                self.request.GET.get("show_errors", "") == "true"
+            )
```

Comment:
So yes technically the request is available on the form and you can query it like this, but this kind of request unpacking really should happen on the view in the `get_form_kwargs` method, and then just be a kwarg on init like

```py
    def __init__(self, *args, show_errors=False, **kwargs):
        super().__init__(*args, **kwargs)
        self.show_errors = show_errors
```